assets/js/components/SelectionPanel/SelectionPanelFooter.js

```diff
@@ -0,0 +1,194 @@
+			setWasSaved( true );
+		}
+	}, [
+		saveSettings,
+		selectedItemSlugs,
+		onSaveSuccess,
+		closeFn,
+		currentButtonText,
+	] );
+
+	const onCancelClick = useCallback( () => {
+		closeFn();
+		onCancel();
+	}, [ closeFn, onCancel ] );
+
+	const [ prevIsOpen, setPrevIsOpen ] = useState( null );
+
+	useEffect( () => {
+		if ( prevIsOpen !== null ) {
+			// If current isOpen is true, and different from prevIsOpen
+			// meaning it transitioned from false to true and it is not
+			// in closing transition, we should reset the button label
+			// locked when save button was clicked.
+			if ( prevIsOpen !== isOpen ) {
+				if ( isOpen ) {
+					setFinalButtonText( null );
+					setWasSaved( false );
+				}
+			}
+		}
+
+		setPrevIsOpen( isOpen );
+	}, [ isOpen, prevIsOpen ] );
+
+	const selectedItemCount = selectedItemSlugs?.length || 0;
+
+	return (
+		<footer className="googlesitekit-selection-panel-footer">
+			{ saveError && <ErrorNotice error={ saveError } /> }
+			<div className="googlesitekit-selection-panel-footer__content">
+				{ haveSettingsChanged && itemLimitError ? (
+					<ErrorNotice
+						error={ {
+							message: itemLimitError,
+						} }
+						noPrefix={
+							selectedItemCount < minSelectedItemCount ||
+							selectedItemCount > maxSelectedItemCount
+						}
+					/>
+				) : (
+					<p className="googlesitekit-selection-panel-footer__item-count">
+						{ createInterpolateElement(
+							sprintf(
+								/* translators: 1: Number of selected items. 2: Maximum number of items that can be selected. */
+								__(
+									'%1$d selected <MaxCount>(up to %2$d)</MaxCount>',
+									'google-site-kit'
+								),
+								selectedItemCount,
+								maxSelectedItemCount
+							),
+							{
+								MaxCount: (
+									<span className="googlesitekit-selection-panel-footer__item-count--max-count" />
+								),
+							}
+						) }
+					</p>
+				) }
+				<div className="googlesitekit-selection-panel-footer__actions">
+					<Button
+						tertiary
+						onClick={ onCancelClick }
+						disabled={ isBusy }
+					>
+						{ __( 'Cancel', 'google-site-kit' ) }
+					</Button>
+					<SpinnerButton
+						onClick={ onSaveClick }
+						isSaving={ isBusy }
+						disabled={
+							selectedItemCount < minSelectedItemCount ||
+							selectedItemCount > maxSelectedItemCount ||
+							isBusy ||
+							( ! isOpen && wasSaved )
+						}
+					>
+						{ finalButtonText || currentButtonText }
+					</SpinnerButton>
+				</div>
+			</div>
+		</footer>
+	);
+}
+
+SelectionPanelFooter.propTypes = {
+	savedItemSlugs: PropTypes.array,
+	selectedItemSlugs: PropTypes.array,
+	saveSettings: PropTypes.func,
+	saveError: PropTypes.object,
+	itemLimitError: PropTypes.string,
+	minSelectedItemCount: PropTypes.number,
+	maxSelectedItemCount: PropTypes.number,
+	isBusy: PropTypes.bool,
+	onSaveSuccess: PropTypes.func,
+	onCancel: PropTypes.func,
+	isOpen: PropTypes.bool,
+	closeFn: PropTypes.func,
```

Comment:
Both `onCancel` and `closeFn` is being called unconditionally in the component, these should be required or a default value provided above maybe? 

Also, should we rename `closeFn` to `onClose` or `handleClose` for consistency?

Reply:
Changed `closeFn` to `closePanel` (in `MetricsSelectionPanel`) and `closeSheet` (in `SideSheet`). I think this sounds more meaningful now, thank you!

I've provided an empty function as default values for both `onCancel` and `closePanel` functions. Thanks!